packages/aws-cdk-lib/aws-ssm/test/parameter.test.ts

```diff
@@ -652,6 +656,100 @@ test('fromLookup will use the SSM context provider to read value during synthesi
   ]);
 });
 
+test('fromLookup will persist value in context by default', async () => {
+  // GIVEN
+  const app = new cdk.App({ context: { [cxapi.NEW_STYLE_STACK_SYNTHESIS_CONTEXT]: false } });
+  const stack = new cdk.Stack(app, 'my-staq', { env: { region: 'us-east-1', account: '12344' } });
+
+  // WHEN
+  const value = ssm.StringParameter.valueFromLookup(stack, 'my-param-name');
+
+  const synth = app.synth();
+
+  const mockCloudExecutable = new MockCloudExecutable({
+    stacks: synth.stacks,
+    missing: synth.manifest.missing,
+  });
+
+  let requestedParameterName: string | undefined;
+  mockCloudExecutable.sdkProvider.stubSSM({
+    getParameter(request) {
+      requestedParameterName = request.Name;
+      return {
+        Parameter: {
+          Value: '123',
+        },
+      };
+    },
+  });
+
+  const cdkToolkit = new CdkToolkit({
+    cloudExecutable: mockCloudExecutable,
+    configuration: mockCloudExecutable.configuration,
+    sdkProvider: mockCloudExecutable.sdkProvider,
+    deployments: new Deployments({ sdkProvider: mockCloudExecutable.sdkProvider }),
+  });
+
+  await mockCloudExecutable.configuration.load();
+
+  await cdkToolkit.assembly();
+
+  await mockCloudExecutable.configuration.saveContext();
+  const config = await new Configuration({ readUserContext: false }).load();
+
+  // THEN
+  expect(value).toEqual('dummy-value-for-my-param-name');
```

Comment:
Confused by why this is true, did you set value to `dummy-value-for-my-param-name` somewhere?

Reply:
See here: https://github.com/aws/aws-cdk/blob/ab940701ce3b423243c657bf10cf1f2aaa9553de/packages/aws-cdk-lib/aws-ssm/README.md?plain=1#L61-L65

Reply:
Also, if you look at the test before this one, a similar assertion is made (line 641)